tests/test_hierarchical.py

```diff
@@ -0,0 +1,822 @@
+) -> None:
+    """Test that collapsing Figure 2 (e) HCM fixture gives Figure 2 (g) fixture."""
+    assert (
+        collapse_HCM(confounder_interference_HCM_pygraphviz)
+        == confounder_interference_collapsed_nxmixedgraph
+    )
+
+
+def test_collapse_instrument(
+    instrument_HCM_pygraphviz: HierarchicalCausalModel,
+    instrument_collapsed_nxmixedgraph: NxMixedGraph,
+) -> None:
+    """Test that collapsing Figure 2 (i) HCM fixture gives Figure 2 (k) fixture."""
+    assert collapse_HCM(instrument_HCM_pygraphviz) == instrument_collapsed_nxmixedgraph
+
+
+# For Algorithm 2
+
+
+def test_parse_Qvar_no_conditional() -> None:
+    """Test parse_Qvar for an unconditioned input Variable."""
+    assert parse_Qvar(Variable("Q_a")) == ("A", set())
+
+
+def test_parse_Qvar_conditional() -> None:
+    """Test parse_Qvar for conditional input Variable."""
+    assert parse_Qvar(Variable("Q_{a|x,y}")) == ("A", {"X", "Y"})
+
+
+def test_parse_Qvar_format_double_conditional() -> None:
+    """Test that a ValueError is raised for inputs with more than one conditional bar."""
+    with pytest.raises(ValueError) as excinfo:
+        parse_Qvar(Variable("Q^{a|b|c}"))
+    assert excinfo.type is ValueError
+
+
+def test_parse_Qvar_format_single_lhs_var() -> None:
+    """Test that a ValueError is raise for inputs with more than one variable on the lhs."""
+    with pytest.raises(ValueError) as excinfo:
+        parse_Qvar(Variable("Q^{ab|c}"))
+    assert excinfo.type is ValueError
+
+
+@pytest.fixture
+def compl_subgraph_HCM() -> HierarchicalCausalModel:
+    """Pytest fixture for HCM with complicated subgraph structure."""
+    HCM = HCM_from_lists(
+        obs_subunits=["A", "B", "C", "Y"],
+        obs_units=["D"],
+        unobs_units=["U"],
+        edges=[
+            ("U", "A"),
+            ("U", "B"),
+            ("U", "C"),
+            ("A", "B"),
+            ("B", "Y"),
+            ("C", "Y"),
+            ("A", "D"),
+            ("D", "C"),
+        ],
+    )
+    return HCM
+
+
+@pytest.fixture
+def confounder_augmented_nxmixedgraph() -> NxMixedGraph:
+    """Pytest fixture for augmented Confounder HCM in Figure 2 (d)."""
+    Qa = Variable("Q_a")
+    Qya = Variable("Q_{y|a}")
+    Qy = Variable("Q_y")
+    return NxMixedGraph.from_edges(undirected=[(Qa, Qya)], directed=[(Qa, Qy), (Qya, Qy)])
+
+
+@pytest.fixture
+def confounder_interference_augmented_nxmixedgraph() -> NxMixedGraph:
+    """Pytest fixture for augmented Confounder Interference HCM in Figure 2 (h)."""
+    Qa = Variable("Q_a")
+    Qya = Variable("Q_{y|a}")
+    Z = Variable("Z")
+    Qy = Variable("Q_y")
+    return NxMixedGraph.from_edges(
+        undirected=[(Qa, Qya)], directed=[(Qa, Qy), (Qya, Qy), (Qa, Z), (Z, Qya)]
+    )
+
+
+@pytest.fixture
+def instrument_augmented_nxmixedgraph() -> NxMixedGraph:
+    """Pytest fixture for augmented Instrument HCM in Figure A2."""
+    Qaz = Variable("Q_{a|z}")
```

Comment:
Make all of these variables global in the test_hierarchical.py module so you don't have to keep re-defining them